android/scripts/make_rust.sh

```diff
@@ -1,52 +1,91 @@
+architectures=(arm64-v8a armeabi x86_64 x86)
+
+if [ ${#target_triples[@]} != ${#architectures[@]} ]; then
+    echo "Number of target_triples has to match number of architectures!"
+    exit 17
+fi
+
+platform="--platform 29"
+release=""
+build_type=debug
+
 if [[ $* == *--release* ]]; then
     # release builds
-    cargo ndk --platform 29 --target x86_64-linux-android build --release
-    cargo ndk --platform 29 --target aarch64-linux-android build --release
-    cargo ndk --platform 29 --target armv7-linux-androideabi build --release
-    cargo ndk --platform 29 --target i686-linux-android build --release
-else
-    # debug builds
-    cargo ndk --platform 29 --target x86_64-linux-android build
-    cargo ndk --platform 29 --target aarch64-linux-android build
-    cargo ndk --platform 29 --target armv7-linux-androideabi build
-    cargo ndk --platform 29 --target i686-linux-android build
+    release="--release"
+    build_type=release
 fi
 
+for target_triple in ${target_triples[@]}; do
+    cargo ndk $platform --target $target_triple build $release
+done
 
 # Linking ###########################################################
 
-echo "Copying .so files to Android project"
+PATH_TO_ANDROID_MAIN=$root/android/core/core/src/main
 
-PATH_TO_ANDROID_LIBRARY="./android/core"
-PATH_TO_ANDROID_MAIN=$PATH_TO_ANDROID_LIBRARY/core/src/main
+echo "PATH_TO_ANDROID_MAIN is $PATH_TO_ANDROID_MAIN"
 
 rm -fr $PATH_TO_ANDROID_MAIN/jniLibs
 mkdir $PATH_TO_ANDROID_MAIN/jniLibs
-mkdir $PATH_TO_ANDROID_MAIN/jniLibs/arm64-v8a
-mkdir $PATH_TO_ANDROID_MAIN/jniLibs/armeabi
-mkdir $PATH_TO_ANDROID_MAIN/jniLibs/x86_64
-mkdir $PATH_TO_ANDROID_MAIN/jniLibs/x86
+for arch in ${architectures[@]}; do
+    mkdir $PATH_TO_ANDROID_MAIN/jniLibs/$arch
+done
 
-if [[ $* == *--release* ]]; then
-    # release
-    cp $(pwd)/target/aarch64-linux-android/release/libcoepi_core.so $PATH_TO_ANDROID_MAIN/jniLibs/arm64-v8a/libcoepi_core.so
-    cp $(pwd)/target/x86_64-linux-android/release/libcoepi_core.so $PATH_TO_ANDROID_MAIN/jniLibs/x86_64/libcoepi_core.so
-    cp $(pwd)/target/armv7-linux-androideabi/release/libcoepi_core.so $PATH_TO_ANDROID_MAIN/jniLibs/armeabi/libcoepi_core.so
-    cp $(pwd)/target/i686-linux-android/release/libcoepi_core.so $PATH_TO_ANDROID_MAIN/jniLibs/x86/libcoepi_core.so
-else
-    # debug
-    cp $(pwd)/target/aarch64-linux-android/debug/libcoepi_core.so $PATH_TO_ANDROID_MAIN/jniLibs/arm64-v8a/libcoepi_core.so
-    cp $(pwd)/target/x86_64-linux-android/debug/libcoepi_core.so $PATH_TO_ANDROID_MAIN/jniLibs/x86_64/libcoepi_core.so
-    cp $(pwd)/target/armv7-linux-androideabi/debug/libcoepi_core.so $PATH_TO_ANDROID_MAIN/jniLibs/armeabi/libcoepi_core.so
-    cp $(pwd)/target/i686-linux-android/debug/libcoepi_core.so $PATH_TO_ANDROID_MAIN/jniLibs/x86/libcoepi_core.so
-fi
+lib_file=libcoepi_core.so
+
+echo "Copying .so files to $PATH_TO_ANDROID_MAIN/jniLibs..."
+
+cnt=${#target_triples[@]}
+limit=$(($cnt - 1))
```

Comment:
I think that it's nice that we're getting rid of repetition, but at the same time this relatively complicated shell logic is adding technical debt IMO. Also, not sure about Windows. We should move this to cargo-make (or similar). https://github.com/Co-Epi/app-backend-rust/issues/121

Reply:
Will look into cargo-make.

Reply:
cargo-make is a powerful tool. It takes time to get familiar with. We should add it when we its features become necessary, not before. Until then, BASH is fine.